opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs

```diff
@@ -0,0 +1,796 @@
+            .and_then(|v| v.parse().map(Duration::from_millis).ok())
+            .unwrap_or(DEFAULT_TIMEOUT);
+
+        PeriodicReaderWithOwnThreadBuilder {
+            interval,
+            timeout,
+            exporter,
+        }
+    }
+
+    /// Configures the intervening time between exports for a [PeriodicReaderWithOwnThread].
+    ///
+    /// This option overrides any value set for the `OTEL_METRIC_EXPORT_INTERVAL`
+    /// environment variable.
+    ///
+    /// If this option is not used or `interval` is equal to zero, 60 seconds is
+    /// used as the default.
+    pub fn with_interval(mut self, interval: Duration) -> Self {
+        if !interval.is_zero() {
+            self.interval = interval;
+        }
+        self
+    }
+
+    /// Configures the timeout for an export to complete. PeriodicReaderWithOwnThread itself
+    /// does not enforce timeout. Instead timeout is passed on to the exporter
+    /// for each export attempt.
+    ///
+    /// This option overrides any value set for the `OTEL_METRIC_EXPORT_TIMEOUT`
+    /// environment variable.
+    ///
+    /// If this option is not used or `timeout` is equal to zero, 30 seconds is used
+    /// as the default.
+    pub fn with_timeout(mut self, timeout: Duration) -> Self {
+        if !timeout.is_zero() {
+            self.timeout = timeout;
+        }
+        self
+    }
+
+    /// Create a [PeriodicReaderWithOwnThread] with the given config.
+    pub fn build(self) -> PeriodicReaderWithOwnThread {
+        PeriodicReaderWithOwnThread::new(self.exporter, self.interval, self.timeout)
+    }
+}
+
+/// A [MetricReader] that continuously collects and exports metric data at a set
+/// interval.
+///
+/// By default, PeriodicReaderWithOwnThread will collect and export data every
+/// 60 seconds. The export time is not counted towards the interval between
+/// attempts. PeriodicReaderWithOwnThread itself does not enforce timeout.
+/// Instead timeout is passed on to the exporter for each export attempt.
+///
+/// The [collect] method of the returned continues to gather and
+/// return metric data to the user. It will not automatically send that data to
+/// the exporter outside of the predefined interval.
+///
+/// As this spuns up own background thread, this is recommended to be used with push exporters
+/// that do not require any particular async runtime. As of now, this cannot be used with
+/// OTLP exporters as they requires async runtime
+///
+/// [collect]: MetricReader::collect
+///
+/// # Example
+///
+/// ```no_run
+/// use opentelemetry_sdk::metrics::PeriodicReaderWithOwnThread;
+/// # fn example<E>(get_exporter: impl Fn() -> E)
+/// # where
+/// #     E: opentelemetry_sdk::metrics::exporter::PushMetricExporter,
+/// # {
+///
+/// let exporter = get_exporter(); // set up a push exporter
+///
+/// let reader = PeriodicReaderWithOwnThread::builder(exporter).build();
+/// # drop(reader);
+/// # }
+/// ```
+#[derive(Clone)]
+pub struct PeriodicReaderWithOwnThread {
```

Comment:
I think we should use a different name. Something more intuitive to the user (component author) like: `PeriodicReaderWithBlockingExporter` or `PeriodicReaderWithSynchronousExporter` or `PeriodicReaderWithNoRuntime`

Reply:
As mentioned in the PR desc, this will be merged into PeriodicReader itself for stable, so don't have to worry about a perfect name.
I'll move this under a feature flag as suggested here: https://github.com/open-telemetry/opentelemetry-rust/pull/2292#discussion_r1835184394